com.unity.perception/Runtime/Randomization/Randomizers/RandomizerExamples/Utilities/PoissonDiskSampling.cs

```diff
@@ -71,6 +85,51 @@ public void Execute()
             }
         }
 
+        /// <summary>
+        /// This job is for filtering out all super sampled Poisson points that are found outside of the originally
+        /// specified 2D region. This job will also shift the cropped points back to their original region.
+        /// </summary>
+        [BurstCompile]
+        struct CropJob : IJob
+        {
+            public float width;
+            public float height;
+            public float minimumRadius;
+            [ReadOnly] public NativeArray<float2> superSampledPoints;
+            public NativeList<float2> croppedSamples;
+
+            public void Execute()
+            {
+                var results = new NativeArray<bool>(
+                    superSampledPoints.Length, Allocator.Temp, NativeArrayOptions.UninitializedMemory);
+
+                // The comparisons operations made in this loop are done separately from the list-building loop
+                // so that burst can automatically generate vectorized assembly code this portion of the job.
+                for (var i = 0; i < superSampledPoints.Length; i++)
+                {
+                    var point = superSampledPoints[i];
+                    results[i] = point.x >= minimumRadius && point.x <= width + minimumRadius
+                        && point.y >= minimumRadius && point.y <= height + minimumRadius;
+                }
+
+                // This list-building code is done separately from the filtering loop
+                // because it cannot be vectorized by burst.
+                for (var i = 0; i < superSampledPoints.Length; i++)
+                {
+                    if (results[i])
+                        croppedSamples.Add(superSampledPoints[i]);
+                }
+
+                // Remove the positional offset from the filtered-but-still-super-sampled points
+                for (var i = 0; i < croppedSamples.Length; i++)
+                {
+                    croppedSamples[i] -= new float2(minimumRadius, minimumRadius);
```

Comment:
Can we create this float2 just once instead of in each iteration of the loop, or would that mess up the burst compiler?

Reply:
Yep. I moved it outside the loop for ya.